### PR TITLE
Config backend runtime attributes. Closes #1315

### LINKED ISSUE #1315: Config backend: runtime attributes
The Config (SFS) backend trait should dynamically fill in the `runtimeAttributeDefinitions` field based on its config.

This will allow the JobPreparation to correctly assign the defaults, and allow the call cache hashing to hash the attributes (if appropriate).

Depends on #1307

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

File: supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigBackendLifecycleActorFactory.scala [hunk 1]
```diff
@@ -26,16 +25,12 @@ class ConfigBackendLifecycleActorFactory(val configurationDescriptor: BackendCon
...
-      RuntimeAttributeDefinition(DockerKey, required = false, None, usedInCallCaching = true),
-      RuntimeAttributeDefinition(ContinueOnReturnCodeKey, required = false, Some(WdlBoolean(false)), usedInCallCaching = true),
-      RuntimeAttributeDefinition(CpuKey, required = false, None, usedInCallCaching = false),
-      RuntimeAttributeDefinition(FailOnStderrKey, required = false, Some(WdlBoolean(true)), usedInCallCaching = true),
-      RuntimeAttributeDefinition(MemoryKey, required = false, None, usedInCallCaching = false)
-    )
+  override def runtimeAttributeDefinitions(initializationDataOption: Option[BackendInitializationData]):
+  Set[RuntimeAttributeDefinition] = {
+    val initializationData = BackendInitializationData.
+      as[SharedFileSystemBackendInitializationData](initializationDataOption)
```
Comment (cjllanwarne) [inline supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigBackendLifecycleActorFactory.scala]: tol: Is pattern matching a safer way of casting this?

**Comment by cjllanwarne** [comment]
LGTM, pending travis 👍

**Comment by mcovarr** [comment]
:+1:

File: supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigBackendLifecycleActorFactory.scala (hunk 1 above)
Comment (kshakir) [inline supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigBackendLifecycleActorFactory.scala]: I'm open to a better underlying implementation of `as[_]`, as the current one was a quick and dirty way of unpacking the `Option` and casting back to the correct type. A richer `as[_]` could provide better error messages than the current `ClassCastException` but as I don't expect it to ever, ever happen, so I shrugged and moved on.

**Pull request merged**